### PR TITLE
libdrm 2.4.115 PyYAML gteq 5.0

### PR DESCRIPTION
**Pull request by Xiaohong-Deng**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

The changes are made due to confidential reasons which will be sent to Chris Co and Jon Slobodzian via email.

###### Change Log  

- PyYAML 3.13 -> 5.2
- libdrm 2.4.114 -> 2.4.115

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

None

###### Links to CVEs  
None

###### Test Methodology

- Pipeline build id: 443106 for PyYAML and its dependents, 443085 for libdrm and its dependents.

Hi @jslobodzian @christopherco @anphel31 me and Suresh want to make sure no regressions can happen with this package upgrade, please suggest the best way to verify that such as any BVT pipelines we can run. Thank you.

### CODE REVIEW

**Review by neha170 — COMMENTED** [review]
Just my thought, building this package along with all those that are a direct dependency of this package, in the buddy build pipelines would be helpful in gaining confidence for the package update.

File: SPECS/PyYAML/PyYAML.spec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Summary:        YAML parser and emitter for Python
 Name:           PyYAML
-Version:        3.13
-Release:        8%{?dist}
+Version:        5.2
```
Comment (reubeno) [inline SPECS/PyYAML/PyYAML.spec:3]: What analysis has been done to confirm that this will not introduce any compatibility risk in the 2.0 LTS distro?

Comment (Xiaohong-Deng) [reply SPECS/PyYAML/PyYAML.spec:3]: Please read the last comment from Vera under the related dependency. The version is chosen based on the research Jon did that is referenced there.

**Comment by Xiaohong-Deng** [comment]
The 4 patches that are to be removed in this PR because they have been merged to upstream in version 5.2. I link the specific commits that correspond to the individual patches here.

change_default_load.patch [Change default loader for yaml.add_constructor](https://github.com/yaml/pyyaml/commit/8d7a78003ae59752cb33e65d54cbc7b37f1fa359)
ConstructorError_fix.patch [Add custom constructors to multiple loaders](https://github.com/yaml/pyyaml/commit/a5394c04a2ef67ee218b49d7d8319f9ed8bcbbe5)
PyYAML-CVE-2017-18342.patch [Deprecate/warn usage of yaml.load(input)](https://github.com/yaml/pyyaml/commit/0cedb2a0697b2bc49e4f3841b8d4590b6b15657e)
PyYAML-lib3-CVE-2017-18342.patch [Apply FullLoader/UnsafeLoader changes to lib3](https://github.com/yaml/pyyaml/commit/857dff153d2c9f238d3a68821262b7764aeb44ea)

**Comment by Xiaohong-Deng** [comment]
The build is successful, please refer to the run ids of the buddy build pipeline. The PR is ready for review.

**Comment by anphel31** [comment]
> The build is successful, please refer to the run ids of the buddy build pipeline. The PR is ready for review.

We should run buddy builds for ARM64 as well. I have triggered these: 
PyYAML and its dependents - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=443155&view=results
libdrm and its dependents - https://dev.azure.com/mariner-org/mariner/_build/results?buildId=443156&view=results

**Review by anphel31 — APPROVED** [review]

**Pull request merged**